pandas/core/generic.py

```diff
@@ -9492,7 +9495,10 @@ def _align_series(
             if is_series:
                 left = self._reindex_indexer(join_index, lidx, copy)
             elif lidx is None or join_index is None:
-                left = self.copy(deep=copy) if copy or copy is None else self
+                if uses_cow:
+                    left = self.copy(deep=copy)
+                else:
+                    left = self.copy(deep=copy) if copy or copy is None else self
```

Comment:
Yeah I agree, but would prefer to keep things separate. We have a couple of other methods where we return self in case of len(index)==0 for example, would like to change them all together if we want to.

Can put up a pr for this first before merging here if you prefer